### PR TITLE
Record a script's completion as a Timestamp. 

### PR DESCRIPTION
**Pull request by leonardr**
Scripts that do nothing but invoke Monitors and CoverageProviders don't get their own timestamps.

This completes https://jira.nypl.org/browse/SIMPLY-1651 and does some of the work for https://jira.nypl.org/browse/SIMPLY-1649. Using the class name as a fallback script name technically completes 1649, but for some scripts the class names are a little vague and I think a more detailed explanation would be better, so I'll tackle that in a separate circulation branch.

### CODE REVIEW

File: scripts.py [hunk 1]
```diff
@@ -2692,6 +2741,8 @@ def register_problem(cls, source, contribution, computed_sort_name, error_messag
 class Explain(IdentifierInputScript):
     """Explain everything known about a given work."""
 
+    name = "Explain everything known about a given work"
+
```
Comment (leonardr) [inline scripts.py]: We could also say that scripts which are only intended to be run from the command line should be `NoTimestampScripts`.

Comment (aslagle) [reply scripts.py]: There might be some that would be useful to keep track of, like a manual directory import, but I agree about this one.

**Comment by leonardr** [comment]
I've flipped the logic, since almost all of our scripts shouldn't have their own timestamps. This does mean some additional work in circulation and metadata to change some of the script superclasses.

File: scripts.py [hunk 2]
```diff
@@ -1618,9 +1652,11 @@ def do_run(self):
         self._db.commit()
 
 
-class WorkPresentationScript(WorkProcessingScript):
+class WorkPresentationScript(TimestampScript, WorkProcessingScript):
     """Calculate the presentation for Work objects."""
 
+    name = "Recalculate the presentation for works that need it."""
```
Comment (aslagle) [inline scripts.py]: I didn't know you could use different numbers of quotes at the beginning and the end of a string.

Comment (leonardr) [reply scripts.py]: You can't really; this is equivalent to `"foo"bar" == "foobar"`. It works by accident.

**Pull request merged**